### ISSUE TITLE
Bugs - Repeated warning messages on metabase.log

### ISSUE THREAD

**Issue by hey-jude** [metabase/metabase#6354]
- My databases: Redshift
- Metabase version: 0.26.2
- Metabase hosting environment: Ubuntu 14.04 on EC2
- Metabase internal database: H2 (default)

Repeatable steps to reproduce the issue
: On every hours, log messages are produced as below.

I think that problem is about time zone.
( KST means Asia/Seoul. )

```
2017-11-09 13:50:00 WARN  sync-timezone:0 - Error syncing database timezone
java.lang.Exception: Unable to parse date string '2017-11-09 13:50:00.002 KST' for database engine 'h2'
	at metabase.driver$make_current_db_time_fn$fn__22975.invoke(driver.clj:305)
	at metabase.driver$fn__22855$G__22707__22862.invoke(driver.clj:34)
	at metabase.sync.sync_metadata.sync_timezone$fn__33284$sync_timezone_BANG___33289$fn__33290.invoke(sync_timezone.clj:20)
	at metabase.sync.sync_metadata.sync_timezone$fn__33284$sync_timezone_BANG___33289.invoke(sync_timezone.clj:13)
	at metabase.sync.sync_metadata$fn__33526$sync_db_metadata_BANG___33531$fn__33532$fn__33533.invoke(sync_metadata.clj:24)
	at metabase.sync.util$do_with_error_handling.invokeStatic(util.clj:115)
	at metabase.sync.util$do_with_error_handling.invoke(util.clj:110)
	at metabase.sync.util$do_with_error_handling.invokeStatic(util.clj:113)
	at metabase.sync.util$do_with_error_handling.invoke(util.clj:110)
	at clojure.core$partial$fn__4759.invoke(core.clj:2514)
	at metabase.driver$fn__22946.invokeStatic(driver.clj:209)
	at metabase.driver$fn__22946.invoke(driver.clj:209)
	at metabase.driver$fn__22920$G__22701__22929.invoke(driver.clj:34)
	at metabase.sync.util$sync_in_context$fn__30918.invoke(util.clj:106)
	at metabase.sync.util$with_db_logging_disabled$fn__30915.invoke(util.clj:98)
	at metabase.sync.util$with_start_and_finish_logging$fn__30910.invoke(util.clj:87)
	at metabase.sync.util$with_sync_events$fn__30907.invoke(util.clj:70)
	at metabase.sync.util$with_duplicate_ops_prevented$fn__30898.invoke(util.clj:49)
	at metabase.sync.util$do_sync_operation.invokeStatic(util.clj:133)
	at metabase.sync.util$do_sync_operation.invoke(util.clj:130)
	at metabase.sync.sync_metadata$fn__33526$sync_db_metadata_BANG___33531$fn__33532.invoke(sync_metadata.clj:23)
	at metabase.sync.sync_metadata$fn__33526$sync_db_metadata_BANG___33531.invoke(sync_metadata.clj:20)
	at metabase.task.sync_databases.SyncAndAnalyzeDatabase.execute(sync_databases.clj:35)
	at org.quartz.core.JobRunShell.run(JobRunShell.java:213)
	at org.quartz.simpl.SimpleThreadPool$WorkerThread.run(SimpleThreadPool.java:557)
Caused by: java.lang.IllegalArgumentException: Invalid format: "2017-11-09 13:50:00.002 KST" is malformed at "KST"
	at org.joda.time.format.DateTimeFormatter.parseDateTime(DateTimeFormatter.java:945)
	at clj_time.format$parse.invokeStatic(format.clj:160)
	at clj_time.format$parse.invoke(format.clj:156)
	at metabase.driver$make_current_db_time_fn$fn__22975.invoke(driver.clj:301)
	... 24 more
```

**Comment by camsaul** [comment]
@senior any idea why Redshift would be including the timezone in the date string? It looks like the date-formatter is expecting a format of `yyyy-MM-dd HH:mm:ss.SSS` for Redshift but for whatever reason it's trying to parse a string that has a timezone at the end

**Comment by senior** [comment]
Wow. there's a few things going on here, I'll try to separate them. The first thing is that this is just a warning. The intent of the code is to try and figure out what timezone the database is in for debugging/troubleshooting purposes. The warning is harmless, just means we couldn't figure out what timezone your database is in.

It shouldn't be returning a timezone for Redshift, I'm using [this](https://github.com/metabase/metabase/blob/master/src/metabase/driver/redshift.clj#L75) SQL string for getting the time, which doesn't include the timezone.  I'm wondering if the driver change or maybe something else has triggered this difference? When I did some reading about Redshift, I thought it always represented times in UTC, unless you set a timezone for the session, so I'm not sure why we're seeing KST here.

The next issue is that even if we were expecting that timezone in our date parser, it would fail to parse 'KST' as JodaTime doesn't recognize that timezone abbreviation. JodaTime has it's list of abbreviations [here](http://www.joda.org/joda-time/timezones.html), looks like for UTC+9, they accept 'ROK'. This is very similar to a problem we have with an Australian timezone https://github.com/metabase/metabase/issues/5869. I just tested parsing a KST timestamp with the new java time library and it appears to recognize KST without any issues. I wrote up a issue https://github.com/metabase/metabase/issues/5909 for making that switch.
